File: order-service/src/main/java/com/daniellaera/orderservice/service/OrderService.java
package com.daniellaera.orderservice.service;

import com.daniellaera.orderservice.dto.OrderDTO;
import com.daniellaera.orderservice.dto.OrderRequest;
import com.daniellaera.orderservice.dto.PagedResponse;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface OrderService {
    OrderDTO createOrder(OrderRequest request, String userEmail);
    OrderDTO getOrderById(Long id);
    List<OrderDTO> getAllOrders();
    List<OrderDTO> getMyOrders(String userEmail);
    PagedResponse<OrderDTO> getMyOrdersPaged(String userEmail, int page, int size);
    PagedResponse<OrderDTO> getAllOrdersPaged(int page, int size);
}

File: shop-ui/src/app/core/services/order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface OrderDto {
  id: number;
  productName: string;
  quantity: number;
  price: number;
  totalAmount: number;
  status: string;
  userEmail: string;
  createdAt: string;
}

export interface PagedResponse<T> {
  content: T[];
  currentPage: number;
  totalPages: number;
  totalElements: number;
  hasNext: boolean;
  hasPrevious: boolean;
}

@Injectable({ providedIn: 'root' })
export class OrderService {
  constructor(private http: HttpClient) {}

  getMyOrders(page = 0, size = 10): Observable<PagedResponse<OrderDto>> {
    return this.http.get<PagedResponse<OrderDto>>(`/orders/my?page=${page}&size=${size}`);
  }

  getAllOrders(page = 0, size = 10): Observable<PagedResponse<OrderDto>> {
    return this.http.get<PagedResponse<OrderDto>>(`/orders?page=${page}&size=${size}`);
  }

  createOrder(productName: string, quantity: number, price: number): Observable<OrderDto> {
    return this.http.post<OrderDto>('/orders', {
      productName, quantity, price
    });
  }
}
